kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/PartitionedTableId.java

```diff
@@ -145,17 +146,18 @@ public TableId getFullTableId() {
     }
 
     @Override
-    public boolean equals(Object obj) {
-        if (obj instanceof PartitionedTableId) {
-            PartitionedTableId that = (PartitionedTableId) obj;
-            return fullTableId.equals(that.fullTableId);
-        }
-        return false;
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        PartitionedTableId that = (PartitionedTableId) o;
+        return Objects.equals(templateSuffix, that.templateSuffix) &&
+                Objects.equals(fullTableId, that.fullTableId);
```

Comment:
prefer to use instanceof instead of getclass. 
essentially you can just add && templateSuffix.equals(that.templateSuffix).

Reply:
I do not think that is a valid solution. This is the effective java way of handling Equals, cause instanceOf is static. For us equality is falsy if the class is not the same.

Reply:
https://github.com/google/guava/wiki/CommonObjectUtilitiesExplained